x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiForm,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiPageBody,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiButton,
+  EuiPanel,
+} from '@elastic/eui';
+
+import { FlashMessages } from '../../../shared/flash_messages';
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { AppLogic } from '../../app_logic';
+
+import {
+  ALLOWED_CHARS_NOTE,
+  META_ENGINE_CREATION_FORM_DOCUMENTATION_DESCRIPTION,
+  META_ENGINE_CREATION_FORM_ENGINE_NAME_LABEL,
+  META_ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER,
+  META_ENGINE_CREATION_FORM_ENGINE_SOURCE_ENGINES_LABEL,
+  META_ENGINE_CREATION_FORM_MAX_SOURCE_ENGINES_WARNING_TITLE,
+  META_ENGINE_CREATION_FORM_META_ENGINE_DESCRIPTION,
+  META_ENGINE_CREATION_FORM_SUBMIT_BUTTON_LABEL,
+  META_ENGINE_CREATION_FORM_TITLE,
+  META_ENGINE_CREATION_TITLE,
+  SANITIZED_NAME_NOTE,
+} from './constants';
+import { MetaEngineCreationLogic } from './meta_engine_creation_logic';
+
+const engineNameToComboBoxOption = (engineName: string): EuiComboBoxOptionOption<string> => ({
+  label: engineName,
+});
+
+const comboBoxOptionToEngineName = (option: EuiComboBoxOptionOption<string>): string =>
+  option.label;
+
+export const MetaEngineCreation: React.FC = () => {
+  const {
+    configuredLimits: {
+      engine: { maxEnginesPerMetaEngine } = { maxEnginesPerMetaEngine: Infinity },
```

Comment:
[Not a change request] Looks like this is an uncovered branch right now, but we can't get rid of it because of our `Partial<ConfiguredLimits>` typing 😞 I'm hoping to fix that here in a follow-up PR so we can remove this fallback

Reply:
How would you suggest I test this?  I'm kinda stumped tbh

Reply:
You could make a one-off test that mocks `setMockValues` to `configuredLimits: { engine: {} }`, but I don't think that's worth it personally - after https://github.com/elastic/kibana/pull/92841 lands we should be able to remove the `=` default totally